unifai/client/twitter.py

```diff
@@ -0,0 +1,180 @@
+        """Receive a message from the queue"""
+        try:
+            return await asyncio.wait_for(self._message_queue.get(), timeout=0.1)
+        except asyncio.TimeoutError:
+            return None
+
+    @ensure_started
+    async def send_message(self, ctx: TwitterMessageContext, reply_messages: List[Message]):
+        """Send a message using the context"""
+        if not reply_messages:
+            return
+            
+        combined_text = "\n".join([msg.content for msg in reply_messages if msg.content])
+        
+        if len(combined_text) > 280:
+            combined_text = combined_text[:277] + "..."
+            
+        try:
+           
+            self.client.create_tweet(
+                text=combined_text,
+                in_reply_to_tweet_id=ctx.tweet_id
+            )
+            logger.info(f"Replied to tweet ID={ctx.tweet_id}")
+        except Exception as e:
+            logger.error(f"Error replying to tweet {ctx.tweet_id}: {e}")
+
+    async def _poll_mentions(self):
+        """Poll for mentions and add them to the queue"""
+        start_time = datetime.datetime.now(datetime.timezone.utc) - datetime.timedelta(seconds=10)
+        since_id = None
+        query = f"(@{self.bot_screen_name} OR to:{self.bot_screen_name}) -is:retweet -is:quote"
+        
+        base_wait_time = self.poll_interval
+        max_wait_time = 900  
+        current_wait_time = base_wait_time
+
+        while not self._stop_event.is_set():
+            try:
+                tweets_resp = self.client.search_recent_tweets(
+                    query=query,
+                    expansions=["author_id"],
+                    tweet_fields=["author_id", "text"],
+                    user_fields=["username", "name", "description"],
+                    start_time=start_time if not since_id else None,
+                    since_id=since_id,
+                    max_results=10,
+                )
+                
+                current_wait_time = base_wait_time
+                
+                if tweets_resp.data:
+                    users = {user.id: user for user in tweets_resp.includes['users']}
+                    
+                    for tweet in reversed(tweets_resp.data):
+                        since_id = max(since_id or 0, tweet.id)
+                        if not tweet.author_id:
+                            continue
+                            
+                        author = users.get(tweet.author_id)
+                        
+                        if author.username.lower() == self.bot_screen_name.lower():
+                            continue
+                            
+                        logger.info(f"Processing tweet {tweet.id} by {author.username}")
+                        
+                        ctx = TwitterMessageContext(
+                            tweet_id=str(tweet.id),
+                            chat_id=str(tweet.id),  
```

Comment:
twitter has a conversation_id which can be used as chat_id: https://docs.tweepy.org/en/stable/expansions_and_fields.html#tweepy.tweet.PUBLIC_TWEET_FIELDS

you also need to expand it to access the field